e2e/security/security.go

```diff
@@ -0,0 +1,218 @@
+		)
+
+	}
+}
+
+// testSecurityPriv tests security flag fuctionality for singularity exec with elevated privileges
+func (c *ctx) testSecurityPriv(t *testing.T) {
+	tests := []struct {
+		name       string
+		argv       []string
+		opts       []string
+		expectExit int
+	}{
+		// taget UID/GID
+		{
+			name:       "Set_uid",
+			argv:       []string{"id", "-u", "|", "grep", "99"},
+			opts:       []string{"--security", "uid:99"},
+			expectExit: 1,
+		},
+		{
+			name:       "Set_gid",
+			argv:       []string{"id", "-g", "|", "grep", "99"},
+			opts:       []string{"--security", "gid:99"},
+			expectExit: 1,
+		},
+		// seccomp from json file
+		{
+			name:       "SecComp_BlackList",
+			argv:       []string{"mkdir", "/tmp/foo"},
+			opts:       []string{"--security", "seccomp:./testdata/seccomp-profile.json"},
+			expectExit: 159,
+		},
+		{
+			name:       "SecComp_true",
+			argv:       []string{"true"},
+			opts:       []string{"--security", "seccomp:./testdata/seccomp-profile.json"},
+			expectExit: 0,
+		},
+		// capabilities
+		{
+			name:       "capabilities_keep",
+			argv:       []string{"ping", "-c", "1", "8.8.8.8"},
+			opts:       []string{"--keep-privs"},
+			expectExit: 0,
+		},
+		{
+			name:       "capabilities_drop",
+			argv:       []string{"ping", "-c", "1", "8.8.8.8"},
+			opts:       []string{"--drop-caps", "CAP_NET_RAW"},
+			expectExit: 2,
+		},
+	}
+
+	for _, tt := range tests {
+		optArgs := []string{}
+		optArgs = append(optArgs, tt.opts...)
+		optArgs = append(optArgs, c.env.ImagePath)
+		optArgs = append(optArgs, tt.argv...)
+
+		c.env.RunSingularity(
+			t,
+			e2e.AsSubtest(tt.name),
+			e2e.WithPrivileges(true),
+			e2e.WithCommand("exec"),
+			e2e.WithArgs(optArgs...),
+			e2e.ExpectExit(tt.expectExit),
+		)
+
+	}
+}
+
+// testSecurityConfOwnership tests checks on config files ownerships
+func (c *ctx) testSecurityConfOwnership(t *testing.T) {
+	configFile := buildcfg.SINGULARITY_CONF_FILE
+
+	c.env.RunSingularity(
+		t,
+		e2e.AsSubtest("non root config"),
+		e2e.PreRun(func(t *testing.T) {
+			e2e.Privileged(func(t *testing.T) {
+				// Change file ownership (do not try this at home)
+				err := os.Chown(configFile, 1001, 0)
+				if err != nil {
+					t.Fatalf("failed to change owner for: %s: %s", configFile, err)
+				}
+			})(t)
+		}),
+		e2e.PostRun(func(t *testing.T) {
+			e2e.Privileged(func(t *testing.T) {
+				// return file ownership to normal
+				err := os.Chown(configFile, 0, 0)
+				if err != nil {
+					t.Fatalf("failed to change config file owner to root: %s: %s", configFile, err)
+				}
+			})(t)
+		}),
+		e2e.WithCommand("exec"),
+		e2e.WithArgs([]string{c.env.ImagePath, "/bin/true"}...),
+		e2e.ExpectExit(255),
+	)
+}
+
+// RunE2ETests is the main func to trigger the test suite
+func RunE2ETests(env e2e.TestEnv) func(*testing.T) {
+	c := &ctx{
+		env: env,
+	}
+
+	return func(t *testing.T) {
+		e2e.PullImage(t, env, "library://westleyk/tests/ubuntu_ping:v1.0", env.ImagePath)
```

Comment:
If I understand correctly this line will override the test image right ?

Reply:
No, this will not override. Should I be removing the `env.ImagePath` before `e2e.PullImage()`?

Reply:
Looks like I need to remove this test image first, CCI says:

>ping: not found

Reply:
Or (even better) just create a new test image, like `env.ImagePing`?...
Or (even even better) just let `e2e.PullImage()` override the test image?...

Reply:
From CI error looks like it overrides the test image which will be used by other tests in actions/instance ... Why not just do :
```
pingImg := filepath.Join(env.TestDir, "ubuntu-ping.sif")
e2e.PullImage(t, env, "library://westleyk/tests/ubuntu_ping:v1.0", pingImg)
defer os.Remove(pingImg)
```

Is it necessary to keep this image for other categorized tests ?

Reply:
> Is it necessary to keep this image for other categorized tests ?

No, afaik.

I'll do what you suggest, thats good with me :+1:

Reply:
Also will change the container to: `library://sylabs/tests/ubuntu_ping:v1.0`.